### PR TITLE
List supported pretraining model names

### PR DESCRIPTION
**Pull request by rasbt**
Right now, it's pretty tedious to find out which model architectures are supported for pretraining. A user has to look that up in the `config.py`. I think it would be convenient to print a list of supported model names. E.g., 

```bash
litgpt pretrain
```

which prints 

```
stablelm-base-alpha-3b
stablelm-base-alpha-7b
stablelm-tuned-alpha-3b
stablelm-tuned-alpha-7b
stablelm-3b-4e1t
stablelm-zephyr-3b
stablecode-completion-alpha-3b
stablecode-completion-alpha-3b-4k
stablecode-instruct-alpha-3b
stable-code-3b
pythia-14m
pythia-31m
pythia-70m
pythia-160m
pythia-410m
pythia-1b
pythia-1.4b
pythia-2.8b
pythia-6.9b
pythia-12b
pythia-70m-deduped
pythia-160m-deduped
pythia-410m-deduped
pythia-1b-deduped
pythia-1.4b-deduped
pythia-2.8b-deduped
pythia-6.9b-deduped
pythia-12b-deduped
dolly-v2-3b
dolly-v2-7b
dolly-v2-12b
RedPajama-INCITE-Base-3B-v1
RedPajama-INCITE-Chat-3B-v1
RedPajama-INCITE-Instruct-3B-v1
RedPajama-INCITE-7B-Base
RedPajama-INCITE-7B-Chat
RedPajama-INCITE-7B-Instruct
RedPajama-INCITE-Base-7B-v0.1
RedPajama-INCITE-Chat-7B-v0.1
RedPajama-INCITE-Instruct-7B-v0.1
falcon-7b
falcon-7b-instruct
falcon-40b
falcon-40b-instruct
falcon-180B
falcon-180B-chat
open_llama_3b
open_llama_7b
open_llama_13b
vicuna-7b-v1.3
vicuna-13b-v1.3
vicuna-33b-v1.3
vicuna-7b-v1.5
vicuna-7b-v1.5-16k
vicuna-13b-v1.5
vicuna-13b-v1.5-16k
longchat-7b-16k
longchat-13b-16k
Nous-Hermes-llama-2-7b
Nous-Hermes-13b
Nous-Hermes-Llama2-13b
Llama-2-7b-hf
Llama-2-7b-chat-hf
Llama-2-13b-hf
Llama-2-13b-chat-hf
Llama-2-70b-hf
Llama-2-70b-chat-hf
Gemma-2b
Gemma-7b
Gemma-2b-it
Gemma-7b-it
FreeWilly2
CodeLlama-7b-hf
CodeLlama-13b-hf
CodeLlama-34b-hf
CodeLlama-70b-hf
CodeLlama-7b-Python-hf
CodeLlama-13b-Python-hf
CodeLlama-34b-Python-hf
CodeLlama-70b-Python-hf
CodeLlama-7b-Instruct-hf
CodeLlama-13b-Instruct-hf
CodeLlama-34b-Instruct-hf
CodeLlama-70b-Instruct-hf
Platypus-30B
Platypus2-7B
Platypus2-13B
Platypus2-70B
Camel-Platypus2-13B
Camel-Platypus2-70B
Stable-Platypus2-13B
Platypus2-70B-instruct
LLaMA-2-7B-32K
phi-1_5
phi-2
Mistral-7B-v0.1
Mistral-7B-Instruct-v0.1
Mixtral-8x7B-v0.1
Mixtral-8x7B-Instruct-v0.1
Mistral-7B-Instruct-v0.2
tiny-llama-1.1b
tiny-llama-1.1b-chat
Llama-2-7b-chat-hf-function-calling-v2
```

So a user can then copy the name & specify a model via

```
litgpt pretrain --model_name tiny-llama-1.1b
```

What do you think?

### CODE REVIEW

**Comment by rasbt** [comment]
PS: If you like this idea, before merging, I probably should scan the docs to make sure we don't use `litgpt pretrain` without a `--config` or `--model_name` anywhere.

**Comment by awaelchli** [comment]
@rasbt I like the initiative to make it easier to discover the model names. The PR currently lists all models, but my concern is that wouldn't it be confusing to list models that aren't meant for pretraining? Many of them are instruct/chat models that we have included in the config list with the intention that they be downloaded to chat or perhaps be further finetuned. For pretraining, I think it is confusing to list these, and it makes the list very long. 

Perhaps we could come up with a way to filter them out, but I wonder if it's worth it. We also need to be careful not to grow the number of boilerplate code in the scripts, so if we decide to add this, perhaps it could move to a utility that we import and add as a one-liner.

**Comment by rasbt** [comment]
I agree that the list seems long, i.e., if we take 

```
falcon-7b
falcon-7b-instruct
falcon-40b
falcon-40b-instruct
falcon-180B
falcon-180B-chat
```

then the instruct and chat variants are redundant since they are the same architecture as the other ones.

On the other hand, it may make sense to pretrain e.g., `falcon-7b-instruct` in certain cases, i.e., if we start with an initial checkpoint. I.e., suppose I have a `falcon-7b-instruct` model and I want to update it with medical knowledge or so. But that would only apply if one used `--initial_checkpoint_dir`. So in this case for `--model_name`, I agree we should shorten the list.

Right now, I think we can catch all instruction-finetuned model by filtering our strings that contain "instruct", "chat", and "-it" (gemma). I can add that if you agree.

**Comment by rasbt** [comment]
Updated it to filter out chat and instruction models as suggested.

**Review by carmocca — COMMENTED** [review]
I find this too ugly in the code for little added value. The pretraining filtering by substring is also very brittle. Could this be improved by docs instead?

**Comment by rasbt** [comment]
> I find this too ugly in the code for little added value.

Do you mean the filtering or the listing of the models in general?

**Comment by carmocca** [comment]
I mean the filtering in the pretraining script. If anything this should be an utility function. But I still worry about the reliability of filtering by "it", "chat" etc

**Comment by rasbt** [comment]
If we have cases where we don't filter enough, I'd say no harm done.
However, I can see your concern. If we filter out models that are not instruction finetuned, that could be a problem. Let's maybe undo that then.

File: litgpt/pretrain.py [hunk 1]
```diff
@@ -91,7 +91,11 @@ def setup(
     if model_config is not None and model_name is not None:
         raise ValueError("Only one of `model_name` or `model_config` can be set.")
     elif model_config is None and model_name is None:
-        model_name = "tiny-llama-1.1b"
+        from litgpt.config import configs
+        options = [f"{config['name']}" for config in configs]
+        print("Please specify --model_name <model_name>. Available values:")
+        print("\n".join(sorted(options)))
```
Comment (carmocca) [inline litgpt/pretrain.py:97]:
```suggestion
        from litgpt.config import name_to_config
        print("Please specify --model_name <model_name>. Available values:")
        print("\n".join(sorted(name_to_config)))
```

File: litgpt/pretrain.py [hunk 2]
```diff
@@ -91,7 +91,11 @@ def setup(
...
+        quit()
```
Comment (carmocca) [inline litgpt/pretrain.py:98]: Can you instead make this an exception?

Comment (rasbt) [reply litgpt/pretrain.py]: Good suggestion, changed it accordingly.

File: litgpt/pretrain.py [hunk 3]
```diff
@@ -91,7 +91,9 @@ def setup(
     if model_config is not None and model_name is not None:
         raise ValueError("Only one of `model_name` or `model_config` can be set.")
     elif model_config is None and model_name is None:
-        model_name = "tiny-llama-1.1b"
+        from litgpt.config import name_to_config
```
Comment (awaelchli) [inline litgpt/pretrain.py:94]: A local import isn't necessary here. Could you move it to the top of the file?

Comment (rasbt) [reply litgpt/pretrain.py:94]: Will do. It's been some time since I opened this PR and don't recall why I did this. I agree it's unnecessary and weird.

**Pull request merged**